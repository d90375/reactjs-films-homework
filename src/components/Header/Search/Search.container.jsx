import React, { useState } from "react";
import { useDispatch } from "react-redux";
import Search from "./Search";
import { fetchData } from "../../../modules/movieListData/movieListActions";

const SearchContainer = () => {
  const dispatch = useDispatch();
  const [queryText, setQueryText] = useState("");

  const onSubmit = (event) => {
    event.preventDefault();
  };

  const handleQueryChanged = (event) => {
    setQueryText(event.target.value);
  };

  const handleKeyQuery = (event) => {
    if (event.key === "Enter") {
      dispatch(fetchData(queryText));
    }
  };

  const handleClickQuery = () => {
    dispatch(fetchData(queryText));
  };

  return (
    <>
      <Search
        onQueryChanged={handleQueryChanged}
        onKeyQuery={handleKeyQuery}
        onClickQuery={handleClickQuery}
        onSubmit={onSubmit}
        queryText={queryText}
      />
    </>
  );
};

export default SearchContainer;
