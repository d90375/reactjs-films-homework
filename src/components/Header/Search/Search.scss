.search__text {
  padding-left: 18px;
  width: 532px;
  height: 65px;
  border-radius: 0.5rem;
  vertical-align: middle;
  border: 4px solid #fff;
  background: url("../../../assets/images/Search_Icon.png") no-repeat scroll 505px 15px;
  background-size: 35px;
  color: #fff;
  font-family: "Helvetica Neue - thin", serif;
  font-size: 28px;
  font-weight: 400;
  line-height: 22px;
  letter-spacing: 3px;

  &::placeholder {
    color: #fff;
    opacity: 1;
  }
}
