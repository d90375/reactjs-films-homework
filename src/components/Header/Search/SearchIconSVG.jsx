import PropTypes from "prop-types";
import React from "react";

const SearchIconSvg = ({ onClickQuery }) => {
  return (
    <>
      <svg
        onClick={onClickQuery}
        className="search__icon"
        xmlns="http://www.w3.org/2000/svg"
        width="1rem"
        height="1rem"
        viewBox="0 0 28 28"
      >
        <g>
          <g>
            <path
              fill="#fff"
              d="M26.3 28l-8.253-8.27a11.037 11.037 0 0 1-6.964 2.466C4.965 22.196 0 17.223 0 11.098 0 4.973 4.958 0 11.083 0c6.118 0 11.084 4.965 11.084 11.098 0 2.625-.912 5.038-2.429 6.934L28 26.3zm-15.217-7.796c2.427 0 4.714-.95 6.432-2.668a9.069 9.069 0 0 0 2.66-6.438 9.04 9.04 0 0 0-2.66-6.439 9.043 9.043 0 0 0-6.432-2.667c-2.427 0-4.72.956-6.432 2.674a9.07 9.07 0 0 0-2.66 6.438c0 2.434.942 4.72 2.66 6.439a9.01 9.01 0 0 0 6.432 2.66z"
            />
          </g>
        </g>
      </svg>
    </>
  );
};

export default SearchIconSvg;

SearchIconSvg.propTypes = {
  onClickQuery: PropTypes.func.isRequired
};
