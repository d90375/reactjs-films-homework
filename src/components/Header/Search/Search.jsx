import React from "react";
import "./Search.scss";
import PropTypes from "prop-types";

const Search = ({ onQueryChanged, queryText }) => {
  return (
    <form className="search" method="post" action="#">
      <label htmlFor="searchId">
        <input onChange={onQueryChanged} value={queryText} className="search__text" required type="text" id="searchId" name="search" placeholder="Search" />
      </label>
    </form>
  );
};

export default Search;

Search.propTypes = {
  onQueryChanged: PropTypes.func.isRequired,
  queryText: PropTypes.string
};

Search.defaultProps = {
  queryText: ""
};
