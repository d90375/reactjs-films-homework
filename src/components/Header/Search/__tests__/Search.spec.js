import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Search from "../Search";

const renderer = new ShallowRenderer();
const setUp = (props) => {
  return renderer.render(<Search {...props} />);
};

describe("Search component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should render Search component", () => {
    expect(component).toMatchSnapshot();
  });
});
