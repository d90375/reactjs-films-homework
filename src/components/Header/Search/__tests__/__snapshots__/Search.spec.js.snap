// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should render Search component 1`] = `
<form
  action="#"
  className="search"
  method="post"
>
  <label
    htmlFor="searchId"
  />
  <input
    className="search__text"
    id="searchId"
    name="search"
    placeholder="the jungle book"
    required={true}
    type="text"
  />
  <span />
</form>
`;
